src/ERC3009.sol

```diff
@@ -36,6 +36,73 @@ abstract contract ERC3009 is IERC3009, StatefulERC712 {
     |                                      External/Public Interactive Functions                                       |
     \******************************************************************************************************************/
 
+    /// @inheritdoc IERC3009
+    function transferWithAuthorization(
+        address from_,
+        address to_,
+        uint256 value_,
+        uint256 validAfter_,
+        uint256 validBefore_,
+        bytes32 nonce_,
+        bytes memory signature_
+    ) external {
+        _transferWithAuthorizationBefore(from_, validAfter_, validBefore_, nonce_);
+
+        _revertIfInvalidSignature(
+            from_,
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
+                        from_,
+                        to_,
+                        value_,
+                        validAfter_,
+                        validBefore_,
+                        nonce_
+                    )
+                )
+            ),
+            signature_
+        );
+
+        _transferWithAuthorizationAfter(from_, to_, value_, nonce_);
+    }
+
+    /// @inheritdoc IERC3009
+    function transferWithAuthorization(
+        address from_,
+        address to_,
+        uint256 value_,
+        uint256 validAfter_,
+        uint256 validBefore_,
+        bytes32 nonce_,
+        bytes32 r_,
+        bytes32 vs_
+    ) external {
+        _transferWithAuthorizationBefore(from_, validAfter_, validBefore_, nonce_);
+
+        _revertIfInvalidSignature(
+            from_,
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
+                        from_,
+                        to_,
+                        value_,
+                        validAfter_,
+                        validBefore_,
+                        nonce_
+                    )
+                )
+            ),
+            abi.encodePacked(r_, vs_)
```

Comment:
I am confused. Why do we need to `abi.encodePacked(r_, vs_)` at all? Is there no (and can there not be be) a `_revertIfInvalidSignature` that takes and uses `r` and `vs` without ever needing to encode?

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/8/commits/b63a1861b99ff78cc8356262b8076badf40154a4